src/averageAges.js

```diff
@@ -55,7 +85,35 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childrenList = people.filter(child => {
+    return onlyWithSon
+      ? people.some(mother => mother.name === child.mother && child.sex === 'm')
+      : people.some(mother => mother.name === child.mother);
+  });
+
+  const ageDiffSum = childrenList.reduce((acc, child) => {
+    const foundMother = people.find(mother =>
+      mother.name === child.mother
+    );
+
+    if (foundMother) {
+      const ageDiff = child.born - foundMother.born;
+
+      return acc + ageDiff;
+    }
+
+    return acc;
+  }, 0);
+
+  return ageDiffSum / childrenList.length;
```

Comment:
```suggestion
  const ageDifferences = childrenList.map(child => {
    const foundMother = people.find(mother => mother.name === child.mother);
    
    return child.born - foundMother.born;
  }, 0);

  return getAverageAge(ageDifferences, childrenList.length);
```

You iterating over the array of children that was filtered before. 
You don't need this condition.
```
    if (foundMother) {
      const ageDiff = child.born - foundMother.born;

      return acc + ageDiff;
    }
```